components/Filter.js

```diff
@@ -49,8 +66,17 @@ export default function Filter({
           <Backdrop closeSelection={closeSelection} />
           <StyledCategoryContainer>
             {[
-              { _id: "uncategorized", name: "Uncategorized" },
-              ...sortedCategories,
+              {
+                _id: "uncategorized",
+                name: "Uncategorized",
+                count: transactionsList.filter(
+                  (transaction) =>
+                    !transaction.category ||
+                    transaction.category === null ||
+                    transaction.category === undefined
+                ).length,
+              },
+              ...categoriesWithCounts,
             ].map((category) => (
```

Comment:
its a bit unexpected to add a default category in the html markup, you could create a variable "categoriesWithUncategorized" before mapping and adding the counts. then you would not have to manually count the uncategorized elements again, since the map will take care of that.

```js
const categoriesWithUncategorized = [{
    _id: "uncategorized",
    name: "Uncategorized",
  }, ...categories
]

const categoiresWithCount = categoriesWithUncategorized.map((category) => { ...
```